jib-core/src/main/java/com/google/cloud/tools/jib/frontend/JavaDockerContextGenerator.java

```diff
@@ -259,8 +264,23 @@ public void generate(Path targetDirectory) throws IOException {
       Path directoryInContext = targetDirectory.resolve(copyDirective.directoryInContext);
       Files.createDirectory(directoryInContext);
 
-      // Copies dependencies.
-      FileOperations.copy(copyDirective.sourceFiles, directoryInContext);
+      // Copies the source files to the directoryInContext.
+      for (LayerEntry layerEntry : copyDirective.layerEntries) {
+        // This resolves the path to copy the source file to in the {@code directory}.
+        // For example, for a 'baseDirectory' of 'target/jib-docker-context', a 'baseExtractionPath'
+        // of '/app/classes', and an 'actualExtractionPath' of
+        // '/app/classes/com/test/HelloWorld.class', the resolved destination would be
+        // 'target/jib-docker-context/com/test/HelloWorld.class'.
```

Comment:
Something feels not right, or the example is not a good one. I know in reality the file should be at `target/jib-docker-context/classes/com/test/HelloWorld.class' (not directly into the root of `target/jib-docker-context`.

Reply:
Ah right, I messed up the example - should be using `jib-docker-context/classes` to be more like the real deal.